invenio_pidstore/resolver.py

```diff
@@ -62,9 +64,13 @@ def resolve(self, pid_value):
 
         if pid.is_deleted():
             obj_id = pid.get_assigned_object(object_type=self.object_type)
-            raise PIDDeletedError(
-                pid,
-                self.object_getter(obj_id) if obj_id else None)
+            try:
+                obj = self.object_getter(obj_id) if obj_id else None
+            except Exception:
+                current_app.logger.exception("Failed to get object {0}".format(
+                    str(obj_id)))
```

Comment:
IMHO you don't need `str` as `__repr__`, `__str__` or `__unicode__` are chosen by `format` automatically.